src/main/java/com/gradle/Utils.java

```diff
@@ -266,6 +267,20 @@ private static Optional<URI> toUri(String scheme, String host, String path) {
         }
     }
 
+    private static <T> Provider<T> reflectionForUseAtConfigurationTime(Provider<T> provider) {
```

Comment:
minor: I would just call it `forUseAtConfigurationTime`. The fact that it uses reflection is an irrelevant implementation detail.